### PR TITLE
authentication selector resource

### PR DESCRIPTION
**Pull request by wesleymccollam**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,9 +3,10 @@
 * `location` property in `resourceLink` object types for all Resources and DataSources. This property will be removed in a future release. ([#195](https://github.com/pingidentity/terraform-provider-pingfederate/pull/195))
 
 ### Resources
+* **New Resource:** `pingfederate_authentication_selector ` ([#198](https://github.com/pingidentity/terraform-provider-pingfederate/pull/198))
```
Comment (henryrecker-pingidentity) [inline CHANGELOG.md:6]: This PR is 199

File: internal/resource/config/authenticationselector/authentication_selector_resource.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+					objectplanmodifier.RequiresReplace(),
+				},
+			},
+			"parent_ref": schema.SingleNestedAttribute{
+				Description: "The reference to this plugin's parent instance. The parent reference is only accepted if the plugin type supports parent instances.",
+				Optional:    true,
+				Attributes:  resourcelink.ToSchemaNoLocation(),
+			},
+			"configuration": pluginconfiguration.ToSchema(),
+			"last_modified": schema.StringAttribute{
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/authenticationselector/authentication_selector_resource.go:84]: I think we can leave read-only modification timestamps like this out of the provider. They tend to cause issues with unexpected plans.

File: internal/resource/config/authenticationselector/authentication_selector_resource.go [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+			},
+			"plugin_descriptor_ref": schema.SingleNestedAttribute{
+				Required:    true,
+				Description: "Reference to the plugin descriptor for this instance. The plugin descriptor cannot be modified once the instance is created.",
+				Attributes:  resourcelink.ToSchemaNoLocation(),
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+			},
+			"parent_ref": schema.SingleNestedAttribute{
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/authenticationselector/authentication_selector_resource.go:78]: Should we allow parents? Not sure if this is the same deal as the inherited attribute we want to remove.

File: internal/resource/config/authenticationselector/authentication_selector_resource.go [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+func (r *authenticationSelectorResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan authenticationSelectorResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var hasObjectErrMap = make(map[error]bool)
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/authenticationselector/authentication_selector_resource.go:198]: It looks like you only ever add to this once, so there isn't really a need for the map

File: internal/resource/config/authenticationselector/authentication_selector_resource.go [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+func (r *authenticationSelectorResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	var plan authenticationSelectorResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var hasObjectErrMap = make(map[error]bool)
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/authenticationselector/authentication_selector_resource.go:284]: Same things here, the map isn't needed currently

File: internal/resource/config/authenticationselector/authentication_selector_resource.go [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+		if hasErr {
+			resp.Diagnostics.AddError("Failed to create an Authentication Selector due to dependent object", err.Error())
+		}
+	}
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createAuthenticationSelectors := client.NewAuthenticationSelector(plan.SelectorId.ValueString(), plan.Name.ValueString(), *pluginDescriptorRef, configuration)
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/authenticationselector/authentication_selector_resource.go:221]: We haven't done this for other resources yet but we could start using a common createAuthenticationSelectorRequest (or something) method that could reduce a little bit of duplication between Create and Update for building the required parameters.

File: internal/resource/config/authenticationselector/authentication_selector_resource.go (hunk 3 above)
Comment (wesleymccollam) [reply internal/resource/config/authenticationselector/authentication_selector_resource.go:78]: I think so.. Maybe we can remove later if needed ¯\_(ツ)_/¯

File: internal/resource/config/authenticationselector/authentication_selector_resource.go (hunk 4 above)
Comment (wesleymccollam) [reply internal/resource/config/authenticationselector/authentication_selector_resource.go:198]: nice catch

File: internal/resource/config/authenticationselector/authentication_selector_resource.go (hunk 6 above)
Comment (wesleymccollam) [reply internal/resource/config/authenticationselector/authentication_selector_resource.go:221]: I've added a JIRA case for this

File: internal/resource/config/authenticationselector/authentication_selector_resource.go (hunk 2 above)
Comment (wesleymccollam) [reply internal/resource/config/authenticationselector/authentication_selector_resource.go]: removed

File: internal/resource/config/authenticationselector/authentication_selector_resource.go (hunk 5 above)
Comment (wesleymccollam) [reply internal/resource/config/authenticationselector/authentication_selector_resource.go:275]: I added configuration to the map now

File: CHANGELOG.md (hunk 1 above)
Comment (wesleymccollam) [reply CHANGELOG.md]: corrected

**Review by henryrecker-pingidentity — APPROVED** [review]

**Pull request merged**